File: src/services/movie-service/movie.mapper.ts
import {IMovie} from "./types/movie.type";

export class MovieMapper {
    public static map(movie: any) : IMovie {
        return {
            identifier: movie.Identifier,
            title: movie.Title,
            slug: movie.Slug,
            year: movie.Year,
            tmdb: movie.TMDB,
            posterUrl: movie.Poster,
            overview: movie.Overview,
            genres: movie.Genres.map((x: any) => {
                return {
                    identifier: x.Identifier,
                    name: x.Name,
                };
            }),
        };
    }
}

File: api/Trackster.Api/Features/Movies/MovieMapper.cs
using Trackster.Api.Data.Records;
using Trackster.Api.Features.Movies.Types;

namespace Trackster.Api.Features.Movies;

public static class MovieMapper
{
    public static Movie Map(MovieRecord movie, List<GenreRecord> genres)
    {
        return new Movie
        {
            Identifier = movie.Identifier,
            Title = movie.Title,
            Year = movie.Year,
            TMDB =  movie.TMDB,
            Poster = movie.Poster,
            Overview = movie.Overview,
            Slug = movie.Slug,
            Genres = genres.ConvertAll(x => new Genre
            {
                Identifier = x.Identifier,
                Name = x.Name
            })
        };
    }
}